src/main/java/org/commcare/suite/model/Detail.java

```diff
@@ -443,4 +459,74 @@ public boolean evaluateFocusFunction(EvaluationContext ec) {
     public XPathExpression getFocusFunction() {
         return focusFunction;
     }
+
+    private boolean templatePathValid(String templatePathProvided) {
+        if (PRINT_TEMPLATE_PROVIDED_VIA_GLOBAL_SETTING.equals(templatePathProvided)) {
+            return true;
+        } else if (templatePathProvided != null) {
+            try {
+                this.derivedPrintTemplatePath =
+                        ReferenceManager.instance().DeriveReference(templatePathProvided).getLocalURI();
+                return true;
+            } catch (InvalidReferenceException e) {
+                System.out.println("Invalid print template path provided for detail with id " + this.id);
+            }
+        }
+        return false;
+    }
+
+    private boolean detailFieldsHaveIds() {
+        for (DetailField field : this.fields) {
+            if (field.getPrintIdentifierRobust() == null) {
+                return false;
```

Comment:
This logic seems backwards to me. Shouldn't it be "if any field has an identifier, return true. Otherwise return false" rather than "If any field doesn't have an ID, return false"?